app/controllers/events/view/settings.js

```diff
@@ -0,0 +1,49 @@
+import Controller from '@ember/controller';
+import { isEqual } from '@ember/utils';
+
+export default Controller.extend({
+
+  actions: {
+    openEventTransferModal(id, name) {
+      this.setProperties({
+        'isEventTransferModalOpen' : true,
+        'confirmEventName'         : '',
+        'eventId'                  : id,
+        'eventName'                : name
+      });
+    },
+    openConfirmEventTransferModal() {
+      const currentInvite = this.model.roleInvites.createRecord({});
+      let { roles } = this.model;
+      for (const role of roles ? roles.toArray() : []) {
+        if (role.name === 'owner') {
+          currentInvite.set('role', role);
+        }
+      }
+      this.setProperties({
+        'isEventTransferModalOpen'        : false,
+        'isConfirmEventTransferModalOpen' : true,
+        'checked'                         : false,
+        currentInvite
+      });
+    },
+    async transferEvent() {
+      try {
+        this.set('isLoading', true);
+        this.currentInvite.set('roleName', 'owner');
+        const currentInviteObj = await this.currentInvite.save();
+        if (isEqual(currentInviteObj, this.currentInvite)) {
+          this.setProperties({
+            'isConfirmEventTransferModalOpen' : false,
+            'checked'                         : false
+          });
+          this.notify.success(this.l10n.t('Owner Role Invite sent successfully.'));
+          this.set('isLoading', false);
```

Comment:
this needs to be done only once after the catch block is over.

Reply:
@CosmicCoder96 updated. Please review